wapitiCore/attack/mod_spring4shell.py

```diff
@@ -0,0 +1,103 @@
+import uuid
+
+from typing import Optional
+from httpx import RequestError
+from wapitiCore.attack.attack import Attack
+from wapitiCore.definitions.spring4shell import NAME, WSTG_CODE
+from wapitiCore.main.log import log_red, logging
+from wapitiCore.net import Request, Response
+
+
+class ModuleSpring4Shell(Attack):
+    """
+    Detect the Spring4Shell vulnerability
+    """
+
+
+    async def must_attack(self, request: Request, response: Optional[Response] = None):
+        if self.finished is True:
+            return False
+        return True
+
+
+    async def _default_request(self, request_url: str, method: str) -> int:
+        request = Request(
+            path=request_url,
+            method=method,
+        )
+        response = await self.crawler.async_send(request, follow_redirects=False)
+        return response.status
+
+
+    async def _attack_spring4shell_url(self, request: Request):
+        payload_unique_id = uuid.uuid4()
+        payload = self._generate_payload(payload_unique_id)
+
+        try:
+            if not await self._check_spring4shell("GET", request, payload):
+                await self._check_spring4shell("POST", request, payload)
+
+        except RequestError:
+            self.network_errors += 1
+            return
+
+
+    async def attack(self, request: Request, response: Optional[Response] = None):
+        await self._attack_spring4shell_url(request)
+
+
+    async def _check_spring4shell(self, method: str, request: Request, payload: str) -> bool:
+        key_payload = payload.split("=")[0]
+        value_payload = payload.split("=")[1]
+
+        if method == "GET":
+            params = request.get_params
+            params.append([key_payload, value_payload])
+
+            malicious_request = Request(
+                path=request.url,
+                method=method,
+                get_params=params,
+            )
+        else:
+            params = request.post_params
+            params.append([key_payload, value_payload])
+
+            malicious_request = Request(
+                path=request.url,
+                method=method,
+                post_params=params,
+            )
```

Comment:
There's a similar pattern in your 2 conditions, here's a shortened version, the boolean variable may need to be shortened tho 
```Python
get_True_post_False = (method=='GET')
params = requests.get_params if get_True_post_False else requests.post_params
params.append([key_payload, value_payload])
            malicious_request = Request(
                path=request.url,
                method=method,
                get_params=params if get_True_post_False else None,
                post_param=params if not get_True_post_False else None
            )
```